dojo/tools/prowler/parser.py

```
@@ -0,0 +1,435 @@
+import csv
+import json
+import logging
+from io import StringIO
+from json.decoder import JSONDecodeError
+
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+
+class ProwlerParser:
+
+    """
+    A parser for Prowler scan results.
+    Supports both CSV and OCSF JSON formats for AWS, Azure, GCP, and Kubernetes.
+    """
+
+    def get_scan_types(self):
+        return ["Prowler Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Prowler Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Prowler scan results in CSV or OCSF JSON format. Supports AWS, Azure, GCP, and Kubernetes scans."
+
+    def get_findings(self, file, test):
+        """Parses the Prowler scan results file (CSV or JSON) and returns a list of findings."""
+        content = file.read()
+        file.seek(0)
+
+        if isinstance(content, bytes):
+            content = content.decode("utf-8")
+
+        # Get file name/path to determine file type
+        file_name = getattr(file, "name", "")
+
+        # Determine file type based on extension
+        if file_name.lower().endswith(".json"):
+            data = self._parse_json(content)
+            findings = self._parse_json_findings(data, test, file_name=file_name)
+        elif file_name.lower().endswith(".csv"):
+            csv_data = self._parse_csv(content)
+            findings = self._parse_csv_findings(csv_data, test, file_name=file_name)
+        else:
+            # Try to detect format from content if extension not recognized
+            try:
+                data = self._parse_json(content)
+                findings = self._parse_json_findings(data, test, file_name=file_name)
+            except (JSONDecodeError, ValueError):
+                csv_data = self._parse_csv(content)
+                findings = self._parse_csv_findings(csv_data, test, file_name=file_name)
```

I think we should throw an error and not try to guess the format.